lib/timeline/TimeStep.js

```diff
@@ -579,12 +579,14 @@ TimeStep.prototype.getLabelMinor = function(date) {
   // noinspection FallThroughInSwitchStatementJS
   switch (this.scale) {
     case 'week':
-      if(this.isMajor() && date.weekday() !== 0){
-          return "";
+      if (this.isMajor() && date.weekday() !== 0) {
+        return "";
+      } else {
+        //default
+        return format && format.length > 0 ? this.moment(date).format(format) : '';  
```

Comment:
This seems a bit complicated to appease the linter. I think it would be better to use the code you had before and just add an lint ignore:

```javascript
  // noinspection FallThroughInSwitchStatementJS
  switch (this.scale) {
    case 'week':
      if (this.isMajor() && date.weekday() !== 0) {
        return "";
      }
    default:  // eslint-disable-line no-fallthrough
      return format && format.length > 0 ? this.moment(date).format(format) : '';
  }
};
```